### PR TITLE
[WIP] [release/8.0][browser] BrowserWebSocket.ReceiveAsync after server initiated close 

### PR DESCRIPTION
**Pull request by ilonatommy**
Follow up for https://github.com/dotnet/runtime/pull/97002.

1) Add tests for receiving messages after closing the socket in 2 situations:
- C# and JS synchronized the websocket states after close and C# knows server closed it (client checked the WS state) -> throw exception.
- Client did not check the WS state, JS got the "onClose" event but it did not get propagated to C# -> receive the buffered messages.
2) Change logic of closing:
- We used to inform C# at once about receiving "onClose" event in JS (setting "closeReceived = true" and changing the state). This lead to a situation where we could get "receive" request from the client when WS was already in "closeReceived" state and we were not leaving a possibility to receive buffered messages.
- Now we let the "receive" event to arrive and leave a promise in JS even when WS got already closed.

### CODE REVIEW

File: src/libraries/System.Net.WebSockets.Client/tests/CloseTest.cs [hunk 1]
```diff
@@ -367,6 +367,61 @@ public async Task CloseOutputAsync_ServerInitiated_CanSend(Uri server)
...
+            catch (WebSocketException e)
+            {
+                Assert.Equal(WebSocketError.InvalidState, e.WebSocketErrorCode);
+                return;
+            }
+            catch (Exception e)
+            {
+                Assert.True(false, $"Unexpected exception: {e}");
+            }
+            Assert.True(false, "Expected WebSocketException not thrown.");
```
Comment (ilonatommy) [inline src/libraries/System.Net.WebSockets.Client/tests/CloseTest.cs:394]: Change to `Assert.Fail` when CI passes.

**Comment by carlossanlop** [comment]
Today is code complete for the April Release. Please merge this before 4pm PT, otherwise it will have to wait until the May Release.

**Comment by pavelsavara** [comment]
> Today is code complete for the April Release. Please get a Tactics approval and merge this before 4pm PT, otherwise it will have to wait until the May Release.

We are not ready to rush it in.

**Comment by ilonatommy** [comment]
Closing in favor of https://github.com/dotnet/runtime/pull/99673.

**Pull request closed without merging**